### PR TITLE
feat(config): simplify config loading

### PR DESCRIPTION
**Pull request by mask-pp**

### CODE REVIEW

File: pkg/rpc/client.go [hunk 1]
```diff
@@ -56,26 +56,25 @@ type ClientConfig struct {
 	L2EngineEndpoint      string
 	JwtSecret             string
 	RetryInterval         time.Duration
-	Timeout               *time.Duration
-	BackOffMaxRetrys      *big.Int
+	Timeout               time.Duration
+	BackOffMaxRetry       uint64
```
Comment (davidtaikocha) [inline pkg/rpc/client.go:60]: Shall we use `Retries`? ref: https://pkg.go.dev/github.com/cenkalti/backoff/v4@v4.1.3#WithMaxRetries

File: proposer/config.go [hunk 2]
```diff
@@ -69,11 +64,11 @@ func NewConfigFromCliContext(c *cli.Context) (*Config, error) {
 		proposeEmptyBlocksInterval = &interval
 	}
 
-	localAddresses := []common.Address{}
+	var localAddresses []common.Address
 	if c.IsSet(flags.TxPoolLocals.Name) {
 		for _, account := range strings.Split(c.String(flags.TxPoolLocals.Name), ",") {
 			if trimmed := strings.TrimSpace(account); !common.IsHexAddress(trimmed) {
-				return nil, fmt.Errorf("invalid account in --txpool.locals: %s", trimmed)
+				return nil, fmt.Errorf("invalid account in --txpool.LocalAddresses: %s", trimmed)
```
Comment (davidtaikocha) [inline proposer/config.go:71]:
```suggestion
				return nil, fmt.Errorf("invalid account in --txpool.locals: %s", trimmed)
```

Looks like the flag name is not changed, so we should still print `--txpool.locals` here.

File: driver/driver.go [hunk 3]
```diff
@@ -36,9 +37,9 @@ type Driver struct {
 	l1HeadSub  event.Subscription
 	syncNotify chan struct{}
 
-	backOffRetryInterval time.Duration
-	ctx                  context.Context
-	wg                   sync.WaitGroup
+	RetryInterval time.Duration
```
Comment (davidtaikocha) [inline driver/driver.go:40]: This value is not initialized in `InitFromConfig`?

Comment (mask-pp) [reply driver/driver.go:40]: Oh, `RetryInterval` should be deleted just use the same name in `*Config` is enough.

**Review by RogerLamTd — APPROVED** [review]

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**